packages/opentelemetry-instrumentation/src/platform/node/instrumentation.ts

```diff
@@ -19,41 +19,57 @@ import * as path from 'path';
 import * as RequireInTheMiddle from 'require-in-the-middle';
 import { satisfies } from 'semver';
 import { InstrumentationAbstract } from '../../instrumentation';
-import { InstrumentationModuleDefinition } from './types';
+import {
+  InstrumentationBaseNode,
+  InstrumentationModuleDefinition
+} from './types';
 import { diag } from '@opentelemetry/api';
 
 /**
  * Base abstract class for instrumenting node plugins
  */
 export abstract class InstrumentationBase<T = any>
   extends InstrumentationAbstract
-  implements types.Instrumentation {
-  private _modules: InstrumentationModuleDefinition<T>[];
+  implements InstrumentationBaseNode<T> {
+  private _modules?: InstrumentationModuleDefinition<any>[];
   private _hooks: RequireInTheMiddle.Hooked[] = [];
   private _enabled = false;
+  private _timer?: NodeJS.Timeout;
 
   constructor(
     instrumentationName: string,
     instrumentationVersion: string,
     config: types.InstrumentationConfig = {}
   ) {
     super(instrumentationName, instrumentationVersion, config);
+    this._timer = setTimeout(() => {
+      throw new Error('You forgot to call loadInstrumentation in constructor');
+    });
+  }
+
+  loadInstrumentation(instrumentationModuleDefinitions:
+                        InstrumentationModuleDefinition<any>
```

Comment:
```suggestion
                        InstrumentationModuleDefinition<T>
```

Reply:
no module definition can have different type than a main, consider the situation when you are patching both modules, they will have different type

Reply:
if differnt types can be used here why is `T` then possible in `_onRequire` which is called for all types passed in here.

Reply:
I think this is already some left overs from past, which can be handled in different PR, this PR is about fixing one thing mainly